### PR TITLE
MINOR: remove unused previousPartition from RoundRobinPartitioner.java

### PR DESCRIPTION
**Pull request by mingdaoy**
[https://github.com/apache/kafka/blob/e6d242113625f20108b64bf54e796fcbcff6d8a7/clie[…]va/org/apache/kafka/clients/producer/RoundRobinPartitioner.java](https://github.com/apache/kafka/blob/e6d242113625f20108b64bf54e796fcbcff6d8a7/clients/src/main/java/org/apache/kafka/clients/producer/RoundRobinPartitioner.java#L40)

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by mingdaoy** [comment]
Hi maintainers, could you please review and approve this workflow when you have a chance? Thanks!

**Review by chia7712 — APPROVED** [review]

**Pull request merged**
